src/application/index.ts

```diff
@@ -67,13 +65,30 @@ class JupyterLab extends Application<ApplicationShell> {
   }
 
   /**
+   * Promise that resolves when state is restored, returning layout description.
+   *
+   * #### Notes
+   * This is just a reference to `shell.restored`.
```

Comment:
Should we call out this fact?  What if the app needs to do something on top of what the shell does?